common/persistence/visibility/elasticsearch/visibility_manager_metrics.go

```diff
@@ -55,212 +57,201 @@ func (m *visibilityManagerMetrics) GetName() string {
-	m.metricClient.IncCounter(metrics.ElasticsearchScanWorkflowExecutionsScope, metrics.ElasticsearchRequests)
+	scope := m.metricClient.Scope(metrics.PersistenceScanWorkflowExecutionsScope, m.visibilityTypeMetricsTag)
 
-	sw := m.metricClient.StartTimer(metrics.ElasticsearchScanWorkflowExecutionsScope, metrics.ElasticsearchLatency)
+	scope.IncCounter(metrics.VisibilityPersistenceRequests)
+	sw := scope.StartTimer(metrics.VisibilityPersistenceLatency)
+	defer sw.Stop()
 	response, err := m.persistence.ScanWorkflowExecutions(request)
-	sw.Stop()
 
-	if err != nil {
-		m.updateErrorMetric(metrics.ElasticsearchScanWorkflowExecutionsScope, err)
-	}
+	m.updateErrorMetric(scope, err)
 
 	return response, err
 }
 
 func (m *visibilityManagerMetrics) CountWorkflowExecutions(request *visibility.CountWorkflowExecutionsRequest) (*visibility.CountWorkflowExecutionsResponse, error) {
-	m.metricClient.IncCounter(metrics.ElasticsearchCountWorkflowExecutionsScope, metrics.ElasticsearchRequests)
+	scope := m.metricClient.Scope(metrics.PersistenceCountWorkflowExecutionsScope, m.visibilityTypeMetricsTag)
 
-	sw := m.metricClient.StartTimer(metrics.ElasticsearchCountWorkflowExecutionsScope, metrics.ElasticsearchLatency)
+	scope.IncCounter(metrics.VisibilityPersistenceRequests)
+	sw := scope.StartTimer(metrics.VisibilityPersistenceLatency)
+	defer sw.Stop()
 	response, err := m.persistence.CountWorkflowExecutions(request)
-	sw.Stop()
 
-	if err != nil {
-		m.updateErrorMetric(metrics.ElasticsearchCountWorkflowExecutionsScope, err)
-	}
+	m.updateErrorMetric(scope, err)
 
 	return response, err
 }
 
 func (m *visibilityManagerMetrics) DeleteWorkflowExecution(request *visibility.VisibilityDeleteWorkflowExecutionRequest) error {
-	m.metricClient.IncCounter(metrics.ElasticsearchDeleteWorkflowExecutionsScope, metrics.ElasticsearchRequests)
+	scope := m.metricClient.Scope(metrics.PersistenceDeleteWorkflowExecutionScope, m.visibilityTypeMetricsTag)
 
-	sw := m.metricClient.StartTimer(metrics.ElasticsearchDeleteWorkflowExecutionsScope, metrics.ElasticsearchLatency)
+	scope.IncCounter(metrics.VisibilityPersistenceRequests)
+	sw := scope.StartTimer(metrics.VisibilityPersistenceLatency)
+	defer sw.Stop()
 	err := m.persistence.DeleteWorkflowExecution(request)
-	sw.Stop()
 
-	if err != nil {
-		m.updateErrorMetric(metrics.ElasticsearchDeleteWorkflowExecutionsScope, err)
-	}
+	m.updateErrorMetric(scope, err)
 
 	return err
 }
 
-func (m *visibilityManagerMetrics) updateErrorMetric(scope int, err error) {
+func (m *visibilityManagerMetrics) updateErrorMetric(scope metrics.Scope, err error) {
+	if err == nil {
+		return
+	}
 	switch err.(type) {
 	case *serviceerror.InvalidArgument:
-		m.metricClient.IncCounter(scope, metrics.ElasticsearchErrBadRequestCounter)
-		m.metricClient.IncCounter(scope, metrics.ElasticsearchFailures)
+		scope.IncCounter(metrics.VisibilityPersistenceInvalidArgument)
+		scope.IncCounter(metrics.VisibilityPersistenceFailures)
```

Comment:
this is called for all possible cases. Why not call it out of the switch clause.